Detectors/MUON/MCH/Tracking/src/TrackAtVertexSpec.cxx

```diff
@@ -0,0 +1,379 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file TrackAtVertexSpec.cxx
+/// \brief Implementation of a data processor to extrapolate the tracks to the vertex
+///
+/// \author Philippe Pillot, Subatech
+
+#include "TrackAtVertexSpec.h"
+
+#include <chrono>
+#include <stdexcept>
+#include <list>
+
+#include <TMath.h>
+#include <TGeoManager.h>
+#include <TGeoGlobalMagField.h>
+
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/Lifetime.h"
+#include "Framework/Output.h"
+#include "Framework/Task.h"
+
+#include "DetectorsBase/GeometryManager.h"
+#include "MathUtils/Cartesian3D.h"
+#include "Field/MagneticField.h"
+#include "MCHBase/ClusterBlock.h"
+#include "MCHBase/TrackBlock.h"
+#include "TrackParam.h"
+#include "TrackExtrap.h"
+
+namespace o2
+{
+namespace mch
+{
+
+using namespace std;
+using namespace o2::framework;
+
+class TrackAtVertexTask
+{
+ public:
+  //_________________________________________________________________________________________________
+  void init(framework::InitContext& ic)
+  {
+    /// Prepare the track extrapolation tools
+
+    LOG(INFO) << "initializing track extrapolation to vertex";
+
+    if (!gGeoManager) {
+      o2::base::GeometryManager::loadGeometry();
+      if (!gGeoManager) {
+        throw runtime_error("cannot load the geometry");
+      }
+    }
+
+    if (!TGeoGlobalMagField::Instance()->GetField()) {
+      auto l3Current = ic.options().get<float>("l3Current");
```

Comment:
You can use a static method ``o2::base::Propagator::initFieldFromGRP("o2sim_grp.root")``, this will guarantee that the field used for the simulation is initialized.

Reply:
Ok but for the moment I don't want to initialize the field used for the simulation but the one corresponding to the run2 data I am using to develop and test the tracking.